src/garage/torch/algos/vpg.py

```diff
@@ -135,85 +141,143 @@ def train_once(self, itr, paths):
+            itr,
+            TrajectoryBatch.from_trajectory_list(self.env_spec, paths),
+            discount=self.discount)
+        return np.mean(undiscounted_returns)
+
+    def _train(self, itr, obs, actions, rewards, baselines, advantages):
+        """Train the algorithm with minibatch.
 
-        self.baseline.fit(paths)
-        return np.mean(average_returns)
+        Args:
+            itr (int): Iteration number.
+            obs (torch.Tensor): Observation from the environment.
+            actions (torch.Tensor): Predicted action.
+            rewards (torch.Tensor): Feedback from the environment.
+            baselines (torch.Tensor): Value function estimation at each step.
+            advantages (torch.Tensor): Expected rewards over the actions.
+
+        Returns:
+            torch.Tensor: Calculated mean value of loss
+
+        """
+        loss = self._compute_loss(itr, obs, actions, rewards, baselines,
+                                  advantages)
+
+        self._optimizer.zero_grad()
+        loss.backward()
+
+        self._optimize(itr, obs, actions, rewards, baselines, advantages)
 
-    def _compute_loss(self, itr, obs, actions, rewards, valids, baselines):
+        return loss
+
+    def _compute_loss(self, itr, obs, actions, rewards, baselines, advantages):
         """Compute mean value of loss.
 
         Args:
             itr (int): Iteration number.
             obs (torch.Tensor): Observation from the environment.
             actions (torch.Tensor): Predicted action.
             rewards (torch.Tensor): Feedback from the environment.
-            valids (list[int]): Array of length of the valid values.
             baselines (torch.Tensor): Value function estimation at each step.
+            advantages (torch.Tensor): Expected rewards over the actions.
 
         Returns:
             torch.Tensor: Calculated mean value of loss
 
         """
-        del itr
+        del itr, baselines
+        objective = self._compute_objective(advantages, obs, actions, rewards)
 
-        policy_entropies = self._compute_policy_entropy(obs)
+        return -objective.mean()
 
-        if self._maximum_entropy:
-            rewards += self._policy_ent_coeff * policy_entropies
+    def _compute_advantage(self, itr, obs, actions, rewards, valids,
+                           baselines):
+        """Compute mean value of loss.
+
+        Args:
+            itr (int): Iteration number.
+            obs (torch.Tensor): Observation from the environment.
+            actions (torch.Tensor): Predicted action.
+            rewards (torch.Tensor): Feedback from the environment.
+            valids (list[int]): Array of length of the valid values.
+            baselines (torch.Tensor): Value function estimation at each step.
+
+        Returns:
+            torch.Tensor: Calculated advantage values given rewards and
+                baselines.
+
+        """
+        del itr, obs, actions
```

Comment:
why take these arguments if you don't need them?

Reply:
I leave these paramesters because these could be used in other (maybe in the future) algorithms. `_compute_objective` in TRPO and VPG is the example.
For VPG `rewards` is not necessary but for other algorithms (TRPO and PPO). If you think there is no possibility to use these parameters later in other algorithms, I will delete them.

There are other parametere like these. I will delete them too if these are also not necessary.

https://github.com/rlworkgroup/garage/blob/b4b5f0a1b676b1eb84deed1b18c574127d9e02ff/src/garage/torch/algos/vpg.py#L186
https://github.com/rlworkgroup/garage/blob/b4b5f0a1b676b1eb84deed1b18c574127d9e02ff/src/garage/torch/algos/vpg.py#L298

Reply:
don't add parameters speculatively. if they are needed in the future, we will add them.

Reply:
fixed